File: 1413-minimum-value-to-get-positive-step-by-step-sum/1413-minimum-value-to-get-positive-step-by-step-sum.java
class Solution {
    public int minStartValue(int[] nums) {
        int [] cumsum = new int[nums.length];
        cumsum[0]= nums[0];
        for(int i=1;i<nums.length;i++)
        {
            cumsum[i] = cumsum[i-1] + nums[i];
        }
        int min = Integer.MAX_VALUE;
        for(int i : cumsum)
            min = Math.min(min,i);
        
        return min < 0 ? Math.abs(min)+1 : 1;
    }
}

File: 1413-minimum-value-to-get-positive-step-by-step-sum/1413-minimum-value-to-get-positive-step-by-step-sum.cpp
class Solution {
public:
    int minStartValue(vector<int>& nums) {
        int sum = 1;
        int n = nums.size();
        int count=1,i=0;
        
        for(; i<n;i++)
        {
            sum += nums[i];         
            if(sum <=0)
            {
                sum = ++count;
                i = -1;
            }
        }
        
        return count;
    }
};